From 0fd5155f934cf002951be2061267d7cc19741506 Mon Sep 17 00:00:00 2001
From: Michal Suchanek <msuchanek@suse.de>
Date: Tue, 10 Oct 2017 16:38:59 +0200
Subject: [PATCH 4/4] powerpc/fadump: use 'fadump_reserve_mem=' when specified

References: bnc#1032084, FATE#323225
Patch-mainline: no, preserve existing SLE behavior in stable update

For stable updates do not change the behavior of existing configuration.
To use crashkernel= for fadump memory reservation remove
fadump_reserve_mem= argument.

Signed-off-by: Michal Suchanek <msuchanek@suse.de>
---
 arch/powerpc/kernel/fadump.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/arch/powerpc/kernel/fadump.c b/arch/powerpc/kernel/fadump.c
index 9c3ce0f8f1bd..4a4f8f26938a 100644
--- a/arch/powerpc/kernel/fadump.c
+++ b/arch/powerpc/kernel/fadump.c
@@ -243,8 +243,12 @@ static inline unsigned long fadump_calculate_reserve_size(void)
 	int ret;
 	unsigned long long base, size;
 
+	/*
+	 * 'fadump_reserve_mem=' is being used to reserve memory
+	 * for firmware-assisted dump.
+	 */
 	if (fw_dump.reserve_bootvar)
-		pr_warn("'fadump_reserve_mem=' parameter is deprecated in favor of 'crashkernel=' parameter.\n");
+		return fw_dump.reserve_bootvar;
 
 	/*
 	 * Check if the size is specified through crashkernel= cmdline
@@ -259,12 +263,6 @@ static inline unsigned long fadump_calculate_reserve_size(void)
 
 		fw_dump.reserve_bootvar = (unsigned long)size;
 		return fw_dump.reserve_bootvar;
-	} else if (fw_dump.reserve_bootvar) {
-		/*
-		 * 'fadump_reserve_mem=' is being used to reserve memory
-		 * for firmware-assisted dump.
-		 */
-		return fw_dump.reserve_bootvar;
 	}
 
 	/* divide by 20 to get 5% of value */
-- 
2.13.6

